pallets/delegation/src/lib.rs

```diff
@@ -130,21 +143,36 @@ pub mod pallet {
 			Payload = Vec<u8>,
 			Signature = Self::Signature,
 		>;
-		type DelegationEntityId: Parameter;
+		// FIXME: Is there a better solution than restrict this to
+		// Into<AccountIdOf<Self>> to enable conversion between both types?
+		type DelegationEntityId: Parameter + Into<AccountIdOf<Self>>;
```

Comment:
there should be no need to convert between the two?
DelegationEntityId -> The DID that creates, revokes and owns Delegatoins
AccountId -> the sender of the TX that pays for everything

Reply:
Yes, that is actually a better point. Not sure why I added it there in the first place, but it would make sense to remove it.

Reply:
f384e39